pkg/sdk/testint/secrets_gen_integration_test.go

```diff
@@ -0,0 +1,673 @@
+		request := sdk.NewCreateWithGenericStringSecretRequest(id, secretString)
+		if with != nil {
+			with(request)
+		}
+		err := client.Secrets.CreateWithGenericString(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupSecret(t, id))
+
+		secret, err := client.Secrets.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		return secret, id
+	}
+
+	assertSecret := func(t *testing.T, s *sdk.Secret, expectedId sdk.SchemaObjectIdentifier, expectedSecretType, expectedComment string) {
+		t.Helper()
+		assert.Equal(t, expectedId.Name(), s.Name)
+		assert.Equal(t, expectedSecretType, s.SecretType)
+		assert.Equal(t, expectedComment, s.Comment)
+		assert.NotEmpty(t, s.CreatedOn)
+		assert.NotEmpty(t, s.DatabaseName)
+		assert.NotEmpty(t, s.SchemaName)
+		assert.NotEmpty(t, s.OwnerRoleType)
+		assert.NotEmpty(t, s.Owner)
+	}
+
+	type secretDetails struct {
+		Name                        string
+		Comment                     string
+		SecretType                  string
+		Username                    string
+		OauthAccessTokenExpiryTime  *time.Time
+		OauthRefreshTokenExpiryTime *time.Time
+		OauthScopes                 string
+		IntegrationName             string
+	}
+
+	assertSecretDetails := func(actual *sdk.SecretDetails, expected secretDetails) {
+		assert.Equal(t, expected.Name, actual.Name)
+		assert.Equal(t, expected.Comment, actual.Comment.String)
+		assert.Equal(t, expected.SecretType, actual.SecretType)
+		assert.Equal(t, expected.Username, actual.Username.String)
+		assert.Equal(t, expected.OauthAccessTokenExpiryTime, actual.OauthAccessTokenExpiryTime)
+		assert.Equal(t, expected.OauthRefreshTokenExpiryTime, actual.OauthRefreshTokenExpiryTime)
+		assert.Equal(t, expected.OauthScopes, actual.OauthScopes.String)
+		assert.Equal(t, expected.IntegrationName, actual.IntegrationName.String)
+	}
+
+	t.Run("Create: secretWithOAuthClientCredentialsFlow", func(t *testing.T) {
+		scopes := []sdk.SecurityIntegrationScope{{Scope: "foo"}, {Scope: "bar"}}
+
+		secret, id := createSecretWithOAuthClientCredentialsFlow(t, integrationId, scopes, func(req *sdk.CreateWithOAuthClientCredentialsFlowSecretRequest) {
+			req.WithComment("a").
+				WithIfNotExists(true)
+		})
+		details, err := client.Secrets.Describe(ctx, id)
+		require.NoError(t, err)
+
+		assertSecretDetails(details, secretDetails{
+			Name:            id.Name(),
+			Comment:         "a",
+			SecretType:      "OAUTH2",
+			OauthScopes:     "[foo, bar]",
+			IntegrationName: integrationId.Name(),
+		})
+
+		assertSecret(t, secret, id, "OAUTH2", "a")
+	})
+
+	// It is possible to create secret without specifying both refresh token properties and scopes
+	// Scopes are not being inherited from the security_integration what is tested further
+	t.Run("Create: secretWithOAuth - minimal, without token and scopes", func(t *testing.T) {
+		secret, id := createSecretWithOAuthClientCredentialsFlow(t, integrationId, []sdk.SecurityIntegrationScope{}, nil)
```

Comment:
That may be just my opinion, but I'm not a big fan of create helper functions in tests that are testing create functionality. Hiding it under the helper function makes it harder to understand (I have to go to the helper function every time). Also, create signature for secrets is not that big, and I wouldn't mind if it was inlined because it would be more clear what parameters exactly are being set. I wouldn't say that those helpers are bad, but they kind of sacrifice the readability which I don't think is worth it. cc: @sfc-gh-jmichalak wdyt?